Userland/Utilities/less.cpp

```diff
@@ -216,8 +217,18 @@ class Pager {
         }
     }
 
+    size_t calculate_digits_count(size_t line_numbers)
+    {
+        return line_numbers > 0 ? (int)log10(line_numbers) + 1 : 1;
+    }
+
     size_t write_range(size_t line, size_t subline, size_t length)
     {
+        char fmt_buf[100];
+        memset(fmt_buf, 0, sizeof(fmt_buf));
+        snprintf(fmt_buf, sizeof(fmt_buf), "{:%lud} ", calculate_digits_count(m_lines.size() + 1));
+        auto fmt_string_size = strlen(fmt_buf);
```

Comment:
Let's not use `snprintf` in C++ when we have `String::formatted`, but none of this is necessary anyways. Format strings support what you are trying to do already - you can pass the width of the string as a format parameter:
```c++
auto digits = calculate_digits_count(m_lines.size() + 1);
outln(m_tty, "{:{}}", i + 1, digits);
```